src/views/Correction.vue

```diff
@@ -1,18 +1,10 @@
 <template>
-  <div>
-    <BenCorrection
-      v-if="isBcBenCccUlc"
-      :correctionFiling="correctionFiling"
-      @fetchError="emitFetchError($event)"
-      @haveData="emitHaveData($event)"
-    />
-    <FmCorrection
-      v-if="isEntityTypeFirm"
-      :correctionFiling="correctionFiling"
-      @fetchError="emitFetchError($event)"
-      @haveData="emitHaveData($event)"
-    />
-  </div>
+  <component
+    :is="component"
+    :correctionFiling="correctionFiling"
+    @fetchError="emitFetchError($event)"
+    @haveData="emitHaveData($event)"
+  />
```

Comment:
This is a good Vue component to know, though we haven't used it much. There's a similar component in Angular and probably all the other UI frameworks.